plugins/woocommerce/client/legacy/js/frontend/woocommerce.js

```diff
@@ -22,31 +31,22 @@ jQuery( function ( $ ) {
 		$( '.woocommerce-store-notice' ).hide();
 	} else {
 		$( '.woocommerce-store-notice' ).show();
-		/**
-		 * After adding the role="button" attribute to the 
-		 * .woocommerce-store-notice__dismiss-link element, 
-		 * we need to add the keydown event listener to it.
-		 */
-		function store_notice_keydown_handler( event ) {
-			if ( ['Enter', ' '].includes( event.key ) ) {
```

Comment:
Enter is handled on anchors by default, no matter the role so it's ok to remove.